Helpers.gs

```diff
@@ -215,13 +214,15 @@ function parseResponses(responses){
       ICAL.TimezoneService.register(tz);
     }
 
+    var colorId = itm[1] && itm[1].colorId
     var allEvents = component.getAllSubcomponents("vevent");
     if (colorId != undefined)
       allEvents.forEach(function(event){event.addPropertyWithValue("color", colorId);});
 
-    var calName = component.getFirstPropertyValue("x-wr-calname") || component.getFirstPropertyValue("name");
-    if (calName != null)
-      allEvents.forEach(function(event){event.addPropertyWithValue("parentCal", calName); });
+    var calName = (itm[1] && itm[1].calendarName) || component.getFirstPropertyValue("x-wr-calname") || component.getFirstPropertyValue("name");
+    if (calName != null) {
+      component.getAllSubcomponents("vevent").forEach(function(event){event.addPropertyWithValue("parentCal", calName); });
+    }
```

Comment:
In preparation of future changes, let's make this less 'hardcoded'.
I'd suggest

```
let overrides = itm[1];
let allEvents = component.getAllSubcomponents("vevent");
let calName = component.getFirstPropertyValue("x-wr-calname") || component.getFirstPropertyValue("name");
if (calName != null)
  allEvents.forEach(function(event){event.addPropertyWithValue("name", calName); });

if (overrides){
  Logger.log(`Applying ${Object.keys(overrides).length} override(s)`);
  allEvents.forEach(function(event){
    for(let overrideProp in overrides){
      event.updatePropertyWithValue(overrideProp, overrides[overrideProp]);
    }
  });
}
```